### PR TITLE
Fixes from comments

### PR DESCRIPTION
**Pull request by joosev**
- Instruction to add shipping methods in Pakettikauppas global settings
- Translations to new texts
- Adds notification if Pakettikauppa plugin is installed but no shipping methods are found in any of the shipping zones
- Fixes PHP error when trying to access array with int when it should access it by string.

### CODE REVIEW

**Comment by ottok** [comment]
Sorry, but these commits are all messed up. Please rebase on master and don't throw away the syntax and other fixes I did yesterday.

image

image

**Comment by joosev** [comment]
Now this pull request only has what I have added, AFAIK.

**Comment by ottok** [comment]
It has now 6 commits instead of 1-2 (or what is the indended changeset). It seems you didn't use [git rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) but something else that added commits.

image

**Comment by joosev** [comment]
So all good?

**Comment by ottok** [comment]
Starting point:
image

`git rebase -i`
```

pick 704f03c rebase changes
s ca8d97a Adds: - Instruction to add shipping methods in Pakettikauppas global settings - Translations to new texts - Adds notification if Pakettikauppa plugin is installed but no shipping methods are found in $
d 06cf6c4 Adds: - Instruction to add shipping methods in Pakettikauppas global settings - Translations to new texts - Adds notification if Pakettikauppa plugin is installed but no shipping methods are found in $
d 45db9f5 Adds: - Instruction to add shipping methods in Pakettikauppas global settings - Translations to new texts - Adds notification if Pakettikauppa plugin is installed but no shipping methods are found in $

# Rebase c8e3d1e..5c51592 onto c8e3d1e (4 commands)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
```

image

Pushed now to pk-master in this repo. Next reviewing changes since the original PR was a mess and I wasn't able to track down in reasonable time what the actual intended changes were.

**Comment by ottok** [comment]
What editor do you use? 
The code is hard to read when the indents jump back and forth. Maybe I can help with editor settings etc so that it would auto-indent?

image

**Comment by ottok** [comment]
Also, your editor should not mix between tabs and spaces:
image

File: includes/class-wc-pakettikauppa.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ public function pickup_point_field_html() {
 
       $chosen_shipping_id = explode( ':', WC()->session->get( 'chosen_shipping_methods' )[0] )[1];
 
-      $shipping_method_id = $shipping_methods[ $chosen_shipping_id ]->instance_settings['shipping_method'];
+      $shipping_method_id = $shipping_methods[ strval($chosen_shipping_id) ]->instance_settings['shipping_method'];
```
Comment (ottok) [inline includes/class-wc-pakettikauppa.php]: Are you sure this is the correct fix to https://github.com/Seravo/woo-pakettikauppa/issues/48 ? Would not the correct fix be to access the array elements by name?

File: includes/class-wc-pakettikauppa-shipping-method.php [hunk 2]
```diff
@@ -56,18 +62,19 @@ public function __construct( $instance_id = 0 ) {
           'instance-settings-modal',
         );
 
-        // Make Pakettikauppa API accessible via WC_Pakettikauppa_Shipment
-        $this->wc_pakettikauppa_shipment = new WC_Pakettikauppa_Shipment();
```
Comment (ottok) [inline includes/class-wc-pakettikauppa-shipping-method.php]: Was the removal of this intentional?

**Comment by ottok** [comment]
Implementation of `wc_pakettikauppa_updated` seems good and logical. I am still looking into the other changes.

There is a new link now to https://docs.woocommerce.com/document/setting-up-shipping-zones/ but I wonder if that is enough. Site owners might struggle to grasp that each shipping method needs to be manually configured for each zone as well.

File: includes/class-wc-pakettikauppa.php (hunk 1 above)
Comment (joosev) [reply includes/class-wc-pakettikauppa.php]: It is accessing the array element by it's name - that is why strval -function call for the chosen_shipping_id (which is int).

Comment (joosev) [reply includes/class-wc-pakettikauppa.php]: All tabs and spaces are created using phpcs.xml file and phpcbf command.

**Comment by joosev** [comment]
> What editor do you use?
> The code is hard to read when the indents jump back and forth. Maybe I can help with editor settings etc so that it would auto-indent?
> 
> image

All tabs and spaces are created using phpcs.xml file and phpcbf command.

**Comment by joosev** [comment]
> What editor do you use?
> The code is hard to read when the indents jump back and forth. Maybe I can help with editor settings etc so that it would auto-indent?
> 
> image

All tabs and spaces are created using phpcs.xml file and phpcbf command.

**Comment by joosev** [comment]
> Implementation of `wc_pakettikauppa_updated` seems good and logical. I am still looking into the other changes.
> 
> There is a new link now to https://docs.woocommerce.com/document/setting-up-shipping-zones/ but I wonder if that is enough. Site owners might struggle to grasp that each shipping method needs to be manually configured for each zone as well.

It should be enough.

**Comment by joosev** [comment]
> > What editor do you use?
> > The code is hard to read when the indents jump back and forth. Maybe I can help with editor settings etc so that it would auto-indent?
> > image
> 
> All tabs and spaces are created using phpcs.xml file and phpcbf command.

I'm using phpcbf because phpcs.xml is not working with Wordpress syntax that my IDE is using.

**Pull request closed without merging**